[Serve] [Docs] Update the "Monitoring Ray Serve" Page

shrekris-anyscale opened the pull request:
<!-- Thank you for your contribution! Please review https://github.com/ray-project/ray/blob/master/CONTRIBUTING.rst before opening a pull request. -->

<!-- Please add a reviewer to the assignee section when you create a PR. If you don't have the access to it, we will shortly find a reviewer and assign them to your PR. -->

## Why are these changes needed?

<!-- Please give a short summary of the change and the problem this solves. -->

The "Monitoring Ray Serve" page explains how to inspect your Ray Serve applications. This change updates the page to remove outdated metrics that Serve no longer exposes and to upgrade code samples to use 2.0 APIs. It also improves the content's readability and organization.

Link to updated "Monitoring Ray Serve" page: https://ray--27777.org.readthedocs.build/en/27777/serve/monitoring.html

## Related issue number

<!-- For example: "Closes #1234" -->

Closes #27720.

## Checks

- [X] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [X] Unit tests
     - This change adds assertions to the documentation code, so they're tested in CI.

shrekris-anyscale commented on doc/source/serve/monitoring.md:
```diff
@@ -128,17 +145,17 @@ clients:
 
-The relevant part for Ray is the `static_configs` field, where we have indicated the location of our log files with `__path__`.
+The relevant part for Ray Serve is the `static_configs` field, where we have indicated the location of our log files with `__path__`.
 The expression `*.*` will match all files, but not directories, which cause an error with Promtail.
```
What does this sentence mean? Is it saying that directories in general cause an error with Promtail? Or is it saying that the expression `*.*` matches all files (but not directories) that cause an error with Promtail?

edoakes replied:
Hmm I'm not actually sure but I'd interpret it as "this matches all files, but doesn't match any directories, which causes an error with Promtail."

I think it was written by @simon-mo or @architkulkarni ?

architkulkarni replied:
I wrote this a while ago, I believe it means that directories cause an error with promtail, so use `*.*` to not match directories

simon-mo replied:
yup @architkulkarni is correct

shrekris-anyscale replied:
Thanks, I revised the wording.

edoakes commented:
@shrekris-anyscale FYI you have a merge conflict here

edoakes commented on doc/source/serve/monitoring.md:
```diff
@@ -2,74 +2,107 @@
+This includes details such as:
+* the number of deployment replicas currently running
+* logs for your Serve controller, deployment replicas, and HTTP proxies
+* the Ray nodes (i.e. machines) running in your Ray cluster
```
period to end the last bullet point

edoakes commented on doc/source/serve/monitoring.md:
````diff
@@ -2,74 +2,107 @@
+* the Ray nodes (i.e. machines) running in your Ray cluster
+
+You can access the Ray dashboard at port 8265 at your cluster's URI.
+For example, if you're running Ray Serve on a local Ray cluster, you can access the dashboard by going to this address in your browser:
+
+```
+http://localhost:8265
+```
````
```suggestion
For example, if you're running Ray Serve locally, you can access the dashboard by going to `http://localhost:8265` in your browser.
```

save some page real estate

edoakes commented on doc/source/serve/monitoring.md:
````diff
@@ -2,74 +2,107 @@
+http://localhost:8265
+```
+
+We can view important information about our application here.
+For example, we can inspect our deployment replicas by navigating to the Ray dashboard's "Actors" tab while our Serve application is running:
````
the previous paragraph used "you" and now you're using "we," stick with one

edoakes reviewed (commented):
Thanks for adding the `serve.run` changes, those will be very helpful.

shrekris-anyscale commented:
@edoakes Thanks for [the review](https://github.com/ray-project/ray/pull/27777#pullrequestreview-1070298557)! I addressed your comments.

dmatrix commented on doc/source/serve/monitoring.md:
```diff
@@ -82,18 +111,21 @@ class Silenced:
-This will prevent the replica INFO-level logs from being written to STDOUT or to files on disk.
+This controls which logs are written to STDOUT or files on disk.
 You can also use your own custom logger, in which case you'll need to configure the behavior to write to STDOUT/STDERR, files on disk, or both.
 
```
Apart from standard Python logger, Serve supports custom logging. Custom logging via configuration can control the behavior of what messages are written to STDOUT/STDERR, files on disk, or both.

dmatrix commented on doc/source/serve/monitoring.md:
```diff
@@ -82,18 +111,21 @@ class Silenced:
-It will be convenient to save the Loki and Promtail executables in the same directory, and to navigate to this directory in your terminal before beginning this walkthrough.
+You can explore and filter your logs using [Loki](https://grafana.com/oss/loki/).
+Setup and configuration is straightforward on Kubernetes, but as a tutorial, let's set up Loki manually.
 
```
Is "setup and configuration" considered a singular compound subject (like ham and cheese is great for breakfast) in this context? If not perhaps change to

"Setup and configuration are straightforward on Kubernetes."

dmatrix commented on doc/source/serve/monitoring.md:
```diff
@@ -163,11 +198,14 @@ You should see something similar to the following:
 :align: center
```
passward "admin."

dmatrix commented on doc/source/serve/monitoring.md:
```diff
@@ -213,16 +247,19 @@ ray start --head --metrics-export-port=8080
 
-The requests will loop and can be canceled with `Ctrl-C`. 
+The requests will loop and can be canceled with `ctrl-c`. 
 
```
The requests will loop until canceled with `ctrl-c`.

dmatrix reviewed (approved):
With those changes, it should be good to go.

Just a general comment. If possible for 2.1, we should consider providing dashboards for this; there is too much boiler code they have to write. But at least we are collecting stats; matter of accessing and presenting easily.

shrekris-anyscale commented:
Thanks for the [review](https://github.com/ray-project/ray/pull/27777#pullrequestreview-1070394225) @dmatrix! I've addressed all your comments.

> Just a general comment. If possible for 2.1, we should consider providing dashboards for this; there is too much boiler code they have to write. But at least we are collecting stats; matter of accessing and presenting easily.

Yeah that's a good suggestion. I think that would improve Ray Serve's observability.
